tls/s2n_resume.c

```diff
@@ -62,6 +62,41 @@ static int s2n_serialize_resumption_state(struct s2n_connection *conn, struct s2
     return 0;
 }
 
+static S2N_RESULT s2n_tls13_serialize_resumption_state(struct s2n_connection *conn, struct s2n_ticket_fields *ticket_fields,
+        struct s2n_stuffer *out)
+{
+    ENSURE_REF(conn);
+    ENSURE_REF(ticket_fields);
+    ENSURE_REF(out);
+
+    uint64_t current_time = 0;
+
+    /* Get the time */
+    GUARD_AS_RESULT(conn->config->wall_clock(conn->config->sys_clock_ctx, &current_time));
+
+    GUARD_AS_RESULT(s2n_stuffer_write_uint8(out, S2N_TLS13_SERIALIZED_FORMAT_VERSION));
+    GUARD_AS_RESULT(s2n_stuffer_write_uint8(out, conn->actual_protocol_version));
+    GUARD_AS_RESULT(s2n_stuffer_write_bytes(out, conn->secure.cipher_suite->iana_value, S2N_TLS_CIPHER_SUITE_LEN));
+    GUARD_AS_RESULT(s2n_stuffer_write_uint64(out, current_time));
+    GUARD_AS_RESULT(s2n_stuffer_write_uint32(out, ticket_fields->ticket_age_add));
+    ENSURE(ticket_fields->session_secret.size <= UINT8_MAX, S2N_ERR_SAFETY);
```

Comment:
Nit: We could use `ENSURE_LTE(ticket_fields->session_secret.size, UINT8_MAX);` here.